### PR TITLE
Fix for maxRetries=0

### LINKED ISSUE #1095: PsrTransport maxRetries off by one
The `$maxRetries` parameter of [`PsrTransport::_construct()`](https://github.com/open-telemetry/opentelemetry-php/blob/ff15cc7037cbdb58641785434b61fd1aa9f27085/src/SDK/Common/Export/Http/PsrTransport.php#L58) appears to be off by one - it's being used as "tries" rather than "retries".

This has the effect of going into an infinite loop if you try and disable retries by passing `0` for `$maxRetries`.

I've created a PR, but this would be a small backwards compatibility break.

### PR DESCRIPTION
**Pull request by stickeegreg**
Fixes #1095

### CODE REVIEW

File: src/SDK/Common/Export/Http/PsrTransport.php [hunk 1]
```diff
@@ -92,7 +92,7 @@ public function send(string $payload, ?CancellationInterface $cancellation = nul
             $request = $request->withAddedHeader($header, $value);
         }
 
-        for ($retries = 0;; $retries++) {
+        for ($tries = 0;; $tries++) {
```
Comment (brettmc) [inline src/SDK/Common/Export/Http/PsrTransport.php:95]: I don't think the variable name change is needed.

**Review by brettmc — CHANGES_REQUESTED** [review]
There's a failing test with this change. Can you look into why, and make that green? I assume the problem here is that if max retire = 0 then it never tries. Does the spec say anything about valid values for maxRetries?

**Comment by stickeegreg** [comment]
Thanks, I've fixed the test and added a new one, also undone renaming the variable.

I had a look at the spec, and as far as I can see it doesn't mention a retry limit at all.
Looking at the Go implementation it seems like they use a maximum time rather than number of tries.

File: src/SDK/Common/Export/Http/PsrTransport.php [hunk 2]
```diff
@@ -110,7 +110,7 @@ public function send(string $payload, ?CancellationInterface $cancellation = nul
                 return new ErrorFuture($e);
             }
 
-            if ($retries + 1 === $this->maxRetries) {
+            if ($retries === $this->maxRetries) {
```
Comment (brettmc) [inline src/SDK/Common/Export/Http/PsrTransport.php:113]: what happens if retries is negative? perhaps this should be a `>=` test...

Comment (stickeegreg) [reply src/SDK/Common/Export/Http/PsrTransport.php]: Tempted to say "you get what you deserve" if you pass a negative maxRetries 😂 
I've pushed up a change so it's `>=` now.

**Review by brettmc — APPROVED** [review]

**Pull request merged**